o11y/o11y.go

```diff
@@ -73,6 +73,16 @@ type Span interface {
 	// https://github.com/open-telemetry/opentelemetry-specification/tree/7ae3d066c95c716ef3086228ef955d84ba03ac88/specification/trace/semantic_conventions
 	AddRawField(key string, val interface{})
 
+	// AddTraceField is for adding application-level information to all spans in the trace
+	// Generally AddField is preferred unless having the field on all spans is useful.
+	//
+	// Any field name will be prefixed with "app."
+	AddTraceField(key string, val interface{})
```

Comment:
Is this any more useful than `AddFieldToTrace` ?

The semantics of `AddFieldToTrace` (taking the context, rather than being called on a single span) - make it more clear that it is going to be added to all spans in the trace

Reply:
It's functionally identical to `AddFieldToTrace` https://github.com/honeycombio/beeline-go/blob/v1.8.0/trace/trace.go#L301, the main difference being that it can be used on the span you're already adding things to.

Honestly I didn't notice that `AddFieldToTrace` was already exposed through this, so no this new function isn't really any more useful then that.

Reply:
One difference I just found is that currently the `AddFieldToTrace` seems to work like `AddField` where it prepends `app.` onto the field being added. I don't immediately see how that's happening in the code, I assumed that it would work like our `AddRawField` 🤔. This isn't' the end of the world, but we do lose some flexibility.

I was hoping to change these to be trace level fields (to make sampling out worker loops easier), but as is I can only safely change the loop_name. https://github.com/circleci/ex/blob/main/worker/worker.go#L83-L84

Reply:
Looks like it's hard coded into the beeline client :( https://github.com/honeycombio/beeline-go/blob/v1.8.0/beeline.go#L303